Source/Core/Core/Boot/Boot.cpp

```diff
@@ -311,20 +309,20 @@ bool CBoot::BootUp()
 		{
 			BS2Success = EmulatedBS2(dolWii);
 		}
-		else if (!VolumeHandler::IsWiiDisc() && !_StartupPara.m_strDefaultISO.empty())
+		else if ((!DVDInterface::VolumeIsValid() || !DVDInterface::GetVolume().IsWiiDisc()) && !_StartupPara.m_strDefaultISO.empty())
```

Comment:
I don't understand what this condition is supposed to be checking, but I added `!DVDInterface::VolumeIsValid()` for safety's sake, because `DVDInterface::GetVolume()` will involve a null pointer dereference if the volume is invalid. `VolumeHandler::IsWiiDisc()` used to return false if the volume is invalid.